coops-ui/src/components/StandaloneDirectorChange/CODDate.vue

```diff
@@ -0,0 +1,241 @@
+     * - the last pre-load Cobrs filing
+     */
+    const lastFilingDate = this.lastFilingDate == null ? 0 : +this.lastFilingDate.split('-').join('')
+    const lastCODFilingDate = this.lastCODFilingDate == null ? 0 : +this.lastCODFilingDate.split('-').join('')
+    const lastPreLoadFilingDate = this.lastPreLoadFilingDate == null ? 0
+      : +this.lastPreLoadFilingDate.split('-').join('')
+    const minCODDate = Math.max(lastFilingDate, lastCODFilingDate, lastPreLoadFilingDate)
+    return this.numToUsableString(minCODDate)
+  }
+
+  /**
+   * Lifecycle hook to load initial data.
+   */
+  private mounted (): void {
+    this.dateFormatted = this.formatDate(this.initialCODDate)
+  }
+
+  /**
+   * Local helper to change date from YYYY-MM-DD to YYYY/MM/DD.
+   * @returns The formatted date.
+   */
+  private formatDate (date: string): string {
+    if (!this.isValidDate(date, '-')) return ''
+    const [year, month, day] = date.split('-')
+    return `${year}/${month}/${day}`
+  }
+
+  /**
+   * Local helper to change date from YYYY/MM/DD to YYYY-MM-DD.
+   * @returns The parsed date.
+   */
+  private parseDate (date: string): string {
+    // changes date from YYYY/MM/DD to YYYY-MM-DD
+    if (!this.isValidDate(date, '/')) return ''
+    const [year, month, day] = date.split('/')
+    return `${year}-${month}-${day}`
+  }
+
+  /**
+   * Local helper to determine if passed-in date is valid.
+   * @returns True if date is valid, otherwise false.
+   */
+  private isValidDate (date, separator): boolean {
+    return (isNotNull.call(this, date) &&
+      isValidFormat.call(this, date, separator) &&
+      isValidCODDate.call(this, date, separator))
+  }
+
+  /**
+   * When prop changes, load (initial) data.
+   */
+  @Watch('initialCODDate')
+  private onInitialCODChanged (val: string): void {
+    if (val) {
+      this.dateFormatted = this.formatDate(val)
+    }
+  }
+
+  /**
+   * When text field changes, update date picker.
+   */
+  @Watch('dateFormatted')
+  private onDateFormattedChanged (val: string): void {
+    this.date = this.parseDate(val)
+  }
+
+  /**
+   * When date picker changes, update text field etc.
+   */
+  @Watch('date')
+  private onDateChanged (val: string): void {
+    const codDate = this.isValidDate(val, '-') ? val : null
+    // only update text field if date is valid
+    // this is to retain previous invalid values
+    if (codDate) {
+      this.dateFormatted = this.formatDate(val)
+    }
+    this.emitCODDate(codDate)
+    this.emitValid(Boolean(codDate))
+  }
+
+  /**
+   * Emits an event with the new value of COD Date.
+   */
+  @Emit('codDate')
+  private emitCODDate (val: string): void { }
+
+  /**
+   * Emits an event indicating whether or not this component is valid.
+   */
+  @Emit('valid')
+  private emitValid (val: boolean): void { }
+}
+
+</script>
+
+<style lang="scss" scoped>
+@import "@/assets/styles/theme.scss";
+
+.validationErrorInfo{
```

Comment:
Please update code style to have a space before the {. This makes it the same code style as in our JS/TS code. I wish our linter would catch this...

Reply:
Resolved in PR #556.